### PR TITLE
Clean up archive before uploading to Pypi

### LINKED ISSUE #2701: Fix PyPI uploading
## Problem Definition

The newly re-enabled PyPI uploading failed due to the presence of the `dist/archive` folder: 
https://travis-ci.org/raiden-network/raiden/jobs/437840495#L4391

Possibly the binary release files will also cause problems, check before fixing.

We can't use `skip_cleanup: false` since otherwise the compiled webui would be deleted as well.

## Solution

- Move PyPI deployment to the end
- Add a script deployment directly before it that deletes the `archive` folder from `dist` after the binary deployment steps are done is done.

### PR DESCRIPTION
**Pull request by rakanalh**
Resolves #2701.

@ulope this implements the solution you've described in the tagged issue. Is this only testable when we want to actually do deployments? or is there a way to test this without triggering an actual release?

### CODE REVIEW

**Comment by ulope** [comment]
@rakanalh You could test it by setting up your raiden fork on travis and registering a test project on PyPI, but that's a lot of work for such a small change. 

One other thing, I mentioned it in the issue description but forgot to include it in the tasklist: From looking at the `dpl` (Travis' deploy tool) source code, it runs `twine upload dist/*`. 

That's a problem because we extract the binary release into the `dist` directory (to run the smoketest on, which we currently don't do because it's broken): https://github.com/raiden-network/raiden/blob/752c9cf7db13cf69a33c331860aecf5d78389f03/.travis.yml#L170

Long story short, we need to change the target of the extraction to prevent twine from trying to upload the binary release file to PyPI.

**Comment by rakanalh** [comment]
@ulope thanks. According to `dpl`'s source code:
https://github.com/travis-ci/dpl/blob/master/lib/dpl/provider/pypi.rb#L124

They run `sdist bdist_wheel` for us, which means all we have to do is redirect everything that is not a build to some other directory. For this PR, we just don't touch `dist` directory and keep it only for use by twine.

File: .travis.yml [hunk 1]
```diff
@@ -177,14 +177,28 @@ jobs:
...
           branch: master
 
+      - provider: script
+        skip_cleanup: true
+        script: .travis/upload_to_s3.sh $DO_S3_ACCESS_KEY $DO_S3_SECRET_KEY build/archive/ raiden-nightlies ams3.digitaloceanspaces.com
+        on:
+          repo: raiden-network/raiden
+          all_branches: true
+
+      - provider: script
```
Comment (ulope) [inline .travis.yml]: With the switch to using `build/` we don't need to delete it anymore, I guess.
